compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -376,7 +379,7 @@ class SpaceEngine(implicit ctx: Context) extends SpaceLogic {
 
   /** Is `tp1` a subtype of `tp2`?  */
   def isSubType(tp1: Type, tp2: Type): Boolean = {
-    val res = tp1 <:< tp2 && (tp1 != ConstantType(Constant(null)) || tp2 == ConstantType(Constant(null)))
+    val res = tp1 <:< tp2 && (tp1 != nullType || tp2 == nullType)
```

Comment:
the nullType checks could be moved before the subtype checks since they're cheaper.